File: javascript/codeeval/167_truncate.js
var fs  = require("fs");

function trunc(inp) {
    if (inp.length > 55) {
        var newStr = inp.substr(0,40)
        var lastSpaceIndex = newStr.lastIndexOf(' ')
        if (lastSpaceIndex) {
            newStr = newStr.substr(0, lastSpaceIndex)
        }
        return newStr + '... <Read More>'
    } else {
        return inp
    }
}
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
        console.log(trunc(line))
    }
});

File: python/codeeval/167_truncate.py
import sys

def trunc(inp):
    inp = inp.rstrip()
    if len(inp) > 55:
        new_str = inp[0:40]
        try:
            last_space = len(new_str) - 1 - new_str[::-1].index(' ')
            new_str = new_str[0:last_space]
        except ValueError:
            pass
        return new_str + '... <Read More>'
    else:
        return inp

with open(sys.argv[1], 'r') as test_cases:
    for test in test_cases:
        print(trunc(test))
